libraries/core_libs/consensus/src/vote_manager/vote_manager.cpp

```diff
@@ -484,14 +484,15 @@ std::optional<VotesBundle> VoteManager::getVotesBundle(uint64_t round, uint64_t
   return votes_bundle;
 }
 
-uint64_t VoteManager::determineRoundFromPeriodAndVotes(uint64_t period, size_t two_t_plus_one) {
+std::optional<std::pair<uint64_t, std::vector<std::shared_ptr<Vote>>>> VoteManager::determineRoundFromPeriodAndVotes(
+    uint64_t period, size_t two_t_plus_one) {
   std::vector<std::shared_ptr<Vote>> votes;
 
   SharedLock lock(verified_votes_access_);
 
   const auto found_period_it = verified_votes_.find(period);
   if (found_period_it == verified_votes_.end()) {
-    return 1;
+    return {};
```

Comment:
I don't really understand why its better to use an optional here, as we can determine that we are in round 1 if we don't have next votes in the period for any round.   And do we we use the votes that are returned?

Reply:
yes we do use votes, which are returned and thats why it is more clear to use optional. Otherwise we would have check if returned round  == 1 and based on that not touch returned votes(which are empty in such case)